File: Codigo-Métodos-Ordenamiento/Python/BubbleSort.py
import random
import time
import os
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]

def read_file(file_path):
    with open(file_path, 'r') as file:
        return [int(line.strip()) for line in file.readlines()]

directory = 'C:/Users/hamdsx/Documents/VScodeExerc/'

execution_times = []

file_names = ['100.txt', '500.txt', '1000.txt', '2000.txt', '3000.txt', 
              '4000.txt', '5000.txt', '6000.txt', '7000.txt', '8000.txt', 
              '9000.txt', '10000.txt', '20000.txt', '30000.txt', 
              '40000.txt', '50000.txt', '60000.txt', '70000.txt', 
              '80000.txt', '90000.txt', '100000.txt']

for file_name in file_names:
    file_path = os.path.join(directory, file_name)
    
    array = read_file(file_path)
    start_time = time.time()
    bubble_sort(array)
    end_time = time.time()
    execution_time = end_time - start_time
    execution_times.append(execution_time)
    print(f"Duración para {file_name}: {execution_time:.10f} segundos.")

print("\nTiempos de ejecución para todos los archivos:")
print(execution_times)

File: Codigo-Métodos-Ordenamiento/Java/BubbleSort/src/main/java/com/mycompany/bubblesort/BubbleSort.java
package com.mycompany.bubblesort;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class BubbleSort {
    public static void main(String[] args) {
        // Cambiar esta línea para especificar la ruta de los archivos
        String basePath = "C:/Users/hamdsx/Documents/VScodeExerc/";
        int totalArchivos = 22; // Total de archivos a analizar
        double[] tiempos = new double[totalArchivos]; // Array para almacenar los tiempos

        // Iterar a través de los archivos
        for (int i = 1; i <= totalArchivos; i++) {
            int[] array = leerArchivo(basePath + i + ".txt"); // Leer archivo
            System.out.println("Array Original para archivo " + i + ": ");
            imprimir(array);  

            long startTime = System.nanoTime(); // Marcar tiempo de inicio
            bubbleSort(array); // Ordenar usando Bubble Sort
            long endTime = System.nanoTime(); // Marcar tiempo de finalización

            tiempos[i - 1] = (endTime - startTime) / 1_000_000_000.0; // Guardar tiempo en segundos
            System.out.println("Duración para archivo " + i + ": " + tiempos[i - 1] + " segundos.");
            System.out.println("Array Ordenado para archivo " + i + ":");
            imprimir(array);
        }

        // Imprimir todos los tiempos al final
        System.out.println("\nTiempos de ordenamiento con Bubble Sort para cada archivo:");
        for (int j = 0; j < tiempos.length; j++) {
            System.out.printf("Archivo %d: %.10f segundos\n", j + 1, tiempos[j]);
        }
    }

    // Método de Bubble Sort
    public static void bubbleSort(int[] array) {
        int size = array.length;
        boolean cambio;
        for (int i = 0; i < size - 1; i++) {
            cambio = false;
            for (int j = 0; j < size - 1 - i; j++) {
                if (array[j] > array[j + 1]) {
                    // Intercambiar array[j] y array[j + 1]
                    int aux = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = aux;
                    cambio = true;
                }
            }
            if (!cambio) break; // Si no hubo intercambio, el array ya está ordenado
        }
    }

    // Método para imprimir el array
    public static void imprimir(int[] array) {
        for (int num : array) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

    // Método para leer el archivo y devolver un array de enteros
    public static int[] leerArchivo(String ruta) {
        // Leer los números del archivo y devolverlos en un array
        int[] numeros = new int[1000000]; // Asumimos un tamaño máximo, ajusta según lo necesario
        int count = 0;

        try (BufferedReader br = new BufferedReader(new FileReader(ruta))) {
            String linea;
            while ((linea = br.readLine()) != null) {
                String[] numerosComoString = linea.split(" ");
                for (String numStr : numerosComoString) {
                    // Parsear y almacenar el número en el array
                    if (count < numeros.length) {
                        numeros[count++] = Integer.parseInt(numStr);
                    }
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        // Crear un nuevo array con el tamaño real
        int[] resultado = new int[count];
        System.arraycopy(numeros, 0, resultado, 0, count);
        return resultado;
    }
}